tests/retrieval/test_metrics.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import pytest
+import torch
+from pytorch_lightning import seed_everything
+from sklearn.metrics import average_precision_score as sk_average_precision
+
+from tests.retrieval.helpers import _assert_error, _compute_sklearn_metric
+from tests.retrieval.helpers import _reciprocal_rank as reciprocal_rank
+from torchmetrics.retrieval.mean_average_precision import RetrievalMAP
+from torchmetrics.retrieval.mean_reciprocal_rank import RetrievalMRR
+
+
+@pytest.mark.parametrize(['sklearn_metric', 'torch_metric'], [
+    [sk_average_precision, RetrievalMAP],
+    [reciprocal_rank, RetrievalMRR],
+])
+@pytest.mark.parametrize('size', [1, 4, 10])
+@pytest.mark.parametrize('rounds', [20])
+@pytest.mark.parametrize('n_documents', [1, 5])
+@pytest.mark.parametrize('query_without_relevant_docs_options', ['skip', 'pos', 'neg'])
+def test_against_sklearn(
+    sklearn_metric, torch_metric, size, rounds, n_documents, query_without_relevant_docs_options
+) -> None:
+    """ Compare PL metrics to standard version. """
+    seed_everything(0)
+
+    for _ in range(rounds):
```

Comment:
why is rounds important, is there a reason that once it can pass but it shall not pass 20x?

Reply:
No, removed